tests/hellinger_distance/test_estimated_hellinger_distance.py

```diff
@@ -249,6 +252,75 @@ def test_train_and_qcompile_with_hellinger_model(source_path: Path, target_path:
         assert predicted_dev.description in get_available_device_names()
 
 
+def test_train_and_qcompile_with_hellinger_model_gnn(source_path: Path, target_path: Path, device: Target) -> None:
```

Comment:
You can use parameterized tests to prevent code duplication (below edit should be done with the original `test_train_and_qcompile_with_hellinger_model` instead).
I think this solution should work for most of the new/original tests.

```suggestion
@pytest.mark.parametrize("gnn", [False, True], ids=["rf", "gnn"])
def test_train_and_qcompile_with_hellinger_model(source_path: Path, target_path: Path, device, gnn: bool) -> None:
...
```